src/linear_algebra/unitary_group.lean

```diff
@@ -0,0 +1,176 @@
+  by { dsimp only, rw [star_star, A.2]}⟩⟩
+
+instance : has_mul (unitary_group n α) :=
+⟨λ A B, ⟨A.1 ⬝ B.1, by rw [star_mul, ←matrix.mul_assoc,
+                                     matrix.mul_assoc _ _ A.1, A.2, matrix.mul_one, B.2],⟩⟩
+
+instance : has_one (unitary_group n α) :=
+⟨⟨1, by rw [star_one, matrix.one_mul]⟩⟩
+
+instance : inhabited (unitary_group n α) := ⟨1⟩
+
+section coe_lemmas
+
+variables (A B : unitary_group n α)
+
+@[simp] lemma inv_val : ↑(A⁻¹) = (star A : matrix n n α) := rfl
+
+@[simp] lemma inv_apply : ⇑(A⁻¹) = (star A : matrix n n α) := rfl
+
+@[simp] lemma mul_val : ↑(A * B) = A ⬝ B := rfl
+
+@[simp] lemma mul_apply : ⇑(A * B) = (A ⬝ B) := rfl
+
+@[simp] lemma one_val : ↑(1 : unitary_group n α) = (1 : matrix n n α) := rfl
+
+@[simp] lemma one_apply : ⇑(1 : unitary_group n α) = (1 : matrix n n α) := rfl
+
+@[simp] lemma to_lin'_mul :
+  to_lin' (A * B) = (to_lin' A).comp (to_lin' B) :=
+matrix.to_lin'_mul A B
+
+@[simp] lemma to_lin'_one :
+  to_lin' (1 : unitary_group n α) = linear_map.id :=
+matrix.to_lin'_one
+
+end coe_lemmas
+
+instance : group (unitary_group n α) :=
+{ mul_assoc := λ A B C, subtype.eq (A.val.mul_assoc B.val C.val),
+  one_mul := λ M, subtype.eq $ one_mul _,
+  mul_one := λ M, subtype.eq $ mul_one _,
+  mul_left_inv := λ A, subtype.eq A.2,
+  ..unitary_group.has_mul,
+  ..unitary_group.has_inv,
+  ..unitary_group.has_one }
+
+/-- `to_linear_equiv A` is matrix multiplication of vectors by `A`, as a linear equivalence. -/
+def to_linear_equiv (A : unitary_group n α) : (n → α) ≃ₗ[α] (n → α) :=
+{ inv_fun := A⁻¹.to_lin',
+  left_inv := λ x, calc
+    A⁻¹.to_lin'.comp A.to_lin' x
+        = (A⁻¹ * A).to_lin' x : by rw [←to_lin'_mul]
+    ... = x : by rw [mul_left_inv, to_lin'_one, id_apply],
+  right_inv := λ x, calc
+    A.to_lin'.comp A⁻¹.to_lin' x
+        = (A * A⁻¹).to_lin' x : by rw [←to_lin'_mul]
+    ... = x : by rw [mul_right_inv, to_lin'_one, id_apply],
+  ..matrix.to_lin' A }
+
+/-- `to_GL` is the map from the unitary group to the general linear group -/
+def to_GL (A : unitary_group n α) : general_linear_group α (n → α) :=
+general_linear_group.of_linear_equiv (to_linear_equiv A)
+
+lemma coe_to_GL (A : unitary_group n α) :
+  (@coe (units _) _ _ (to_GL A)) = A.to_lin' :=
+rfl
+
+@[simp]
+lemma to_GL_one : to_GL (1 : unitary_group n α) = 1 :=
+by { ext1 v i, rw [coe_to_GL, to_lin'_one], refl }
+
+@[simp]
+lemma to_GL_mul (A B : unitary_group n α) :
+  to_GL (A * B) = to_GL A * to_GL B :=
+by { ext1 v i, rw [coe_to_GL, to_lin'_mul], refl }
+
+/-- `unitary_group.embedding_GL` is the embedding from `unitary_group n α`
+  to `general_linear_group n α`. -/
+def embedding_GL : (unitary_group n α) →* (general_linear_group α (n → α)) :=
+⟨λ A, to_GL A, by simp, by simp⟩
+
+end unitary_group
+
+section orthogonal_group
+
+variables (n : Type u) [fintype n] [decidable_eq n]
+variables (β : Type v) [comm_ring β]
+
+abbreviation orthogonal_group := @unitary_group n _ _ β _ star_ring_of_comm
```

Comment:
@semorrison, can you think of any harm caused by making `star_ring_of_comm` a global instance?